### PR TITLE
backend/accounts: de-duplicate account config fields

### PR DESCRIPTION
**Pull request by benma**
accounts.AccountConfig and config.Account have lots of duplicate
fields. Code can be simplifed by reusing the latter in the former.

### CODE REVIEW

**Comment by benma** [comment]
@Beerosagos sorry it's a bit long, but 95% of it are mechanical renames/replacements. I only needed to pay special attention to:

- `Active` in AccountsConfig is `Inactive` in config.Account, so needed to invert everywhere.

File: backend/accounts.go [hunk 1]
```diff
@@ -634,6 +638,7 @@ func (backend *Backend) initPersistedAccounts() {
 	persistedAccounts := backend.config.AccountsConfig()
 outer:
 	for _, account := range backend.filterAccounts(&persistedAccounts, keystoreConnected) {
+		account := account
```
Comment (Beerosagos) [inline backend/accounts.go:641]: Is this needed to make a deep copy of `account` to be passed to `createAndAddAccount`?

**Review by Beerosagos — APPROVED** [review]
tACK!

Small nit: maybe rename `Account.Configurations` (https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/backend/config/accounts.go#L33), since calling `account.Config().Config.Configurations` is not really intuitive

File: backend/accounts.go (hunk 1 above)
Comment (benma) [reply backend/accounts.go:641]: It's not a deep copy. Account is a pointer.

I did it fearing the old pitfall of Go where loop vars are reused. See https://go.dev/play/p/nX3LCI2yqLR.

Some more detail: https://medium.com/swlh/use-pointer-of-for-range-loop-variable-in-go-3d3481f7ffc9

But it seems if the var itself is a pointer there is no issue? https://go.dev/play/p/6Ej6pvOJ3Ci Can't say I understand it. One of the worst pitfalls of Go, and one that is planned to be fixed afaik.

**Comment by benma** [comment]
> tACK!
> 
> Small nit: maybe rename `Account.Configurations` (https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/backend/config/accounts.go#L33), since calling `account.Config().Config.Configurations` is not really intuitive

Yeah it looks bad.

To what though? It is called `(Signing)Configurations` everywhere... I'll rename it to `SigningConfigurations` I guess. 
Maybe the second `.Config` could also be renamed, but not sure to what.

**Comment by Beerosagos** [comment]
> > tACK!
> > Small nit: maybe rename `Account.Configurations` (https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/backend/config/accounts.go#L33), since calling `account.Config().Config.Configurations` is not really intuitive
> 
> Yeah it looks bad.
> 
> To what though? It is called `(Signing)Configurations` everywhere... I'll rename it to `SigningConfigurations` I guess. Maybe the second `.Config` could also be renamed, but not sure to what.

Yeah I was thinking about `SigningConfiguration` too. About the second `Config` would be nice to rename but don't know how, maybe `PersistedConfig`? Not super convinced though..

File: backend/accounts.go (hunk 1 above)
Comment (Beerosagos) [reply backend/accounts.go:641]: Yes, I think we could avoid that. Until we don't use the address of the iteration variable (`&account`) everything should be fine.

**Comment by benma** [comment]
> 

Done the rename now. PersistedConfig sounded good in theory but in practice the lines became very long, easily going beynd the 100 char column width:

`			accountNumber, err := accountConfig.PersistedConfig.SigningConfigurations[0].AccountNumber()`

so i didn't do this one

**Pull request merged**